File: src/codes/Users_Manual/advanced/custom_modulation_0.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Utils;
~using AUTD3Sharp.Gain;
~using static AUTD3Sharp.Units;
using AUTD3Sharp.Derive;
~var m = new Burst(4000);

[Modulation]
public partial class Burst
{
    private readonly int _length;

    public Burst(int length)
    {
        _length = length;
        _config = 4000u * Hz;
    }

    private byte[] Calc()
    {
        var buf = Enumerable.Repeat<byte>(0x00, _length).ToArray();
        buf[0] = 0xFF;
        return buf;
    }
}

File: src/codes/Users_Manual/advanced/custom_modulation_0.py
import numpy as np
from pyautd3 import EmitIntensity, Hz, SamplingConfig
from pyautd3.modulation import Modulation


class Burst(Modulation):
    _length: int

    def __init__(self, length: int = 4000):
        super().__init__(SamplingConfig.Freq(4000 * Hz))
        self._length = length

    def calc(self):
        buf = np.array([EmitIntensity.minimum()] * self._length)
        buf[0] = EmitIntensity.maximum()
        return buf
